signer/mar/README.rst

```diff
@@ -0,0 +1,110 @@
+MAR Signing
+===========
+
+.. sectnum::
+.. contents:: Table of Contents
+
+MAR is Firefox's update file format. For a full description, see
+`go.mozilla.org/mar`_.
+
+.. _`go.mozilla.org/mar`: https://godoc.org/go.mozilla.org/mar
+
+
+Configuration
+-------------
+
+The only thing needed to configure a MAR signer is a private key, either RSA or
+ECDSA P256/P384.
+
+.. code:: yaml
+
+	signers:
+    - id: testmar
+      type: mar
+      privatekey: |
+          -----BEGIN PRIVATE KEY-----
+          MIIJQgIBADANBgkqhkiG9w0BAQEFAASCCSwwggkoAgEAAoICAQDHV+bKFLr1p5FR
+		  ...
+          -----END PRIVATE KEY-----
+
+Signature request
+-----------------
+
+This signer supports `/sign/hash`, `/sign/data/` and `/sign/file` endpoints.
+They all use the same request format:
+
+.. code:: json
+
+	[
+		{
+			"input": "Y2FyaWJvdW1hdXJpY2UK",
+			"keyid": "testmar"
+		}
+	]
+
+The `/sign/file` endpoint takes a whole MAR encoded in base64. It will parse the
+mar, sign it and return the signed file.
+
+The `/sign/data` and `/sign/hash` endpoint only does the signing step. They
+takes a MAR block already prepared for signature, calculate its digest (if
+`/sign/data`) and return the signature bytes to be inserted in the signature
+field. Because the signer needs to know which algorithm to use for signature,
+the signature algorithm can be specified in the signing request options.
+
+.. code:: json
+
+	[
+		{
+			"input": "Y2FyaWJvdW1hdXJpY2UK",
+			"keyid": "testmar",
+			"options": {
+				"sigalg": 1
```

Comment:
nit: nice to link to where these are enumerated

Reply:
added to 672bf7130d886fda16811c4bdeefd35b1c99ebf3